### PR TITLE
Improve test coverage and simplify the queryCloserPeersOnFrontier function

### PR DESCRIPTION
**Pull request by tok-kkk**

### CODE REVIEW

File: node_test.go [hunk 1]
```diff
@@ -149,4 +153,125 @@ var _ = Describe("Bootstrapping", func() {
...
+						log.Println(err)
+					}
+				}
+				log.Printf("%v/%v successful pings", numberOfPings, 20)
+				Ω(numberOfPings).Should(BeNumerically(">=", 2*20/3))
+			})
+		})
+
+		Context("context", func() {
+			It("should return an error when calling with a canceled context", func() {
```
Comment (jazg) [inline node_test.go]: cancelled*

File: node_test.go [hunk 2]
```diff
@@ -149,4 +153,125 @@ var _ = Describe("Bootstrapping", func() {
...
+			})
+		})
+
+		Context("pruning the dht", func() {
+			It("should return an error when pruning with bad dht", func() {
+				keyPair, err := identity.NewKeyPair()
+				Ω(err).ShouldNot(HaveOccurred())
+
+				correctAddress := keyPair.Address()
+				worngFormatTarget := identity.Address("wrongAddress")
```
Comment (jazg) [inline node_test.go]: wrongFormatTarget*

File: node_test.go [hunk 3]
```diff
@@ -149,4 +153,125 @@ var _ = Describe("Bootstrapping", func() {
...
+
+			It("should return an error with a bad query input", func() {
+				_, err := node.QueryCloserPeers(context.Background(), &rpc.Query{Query: &rpc.Address{}, From: &rpc.MultiAddress{}})
+				Ω(err).Should(HaveOccurred())
+
+			})
+		})
+
+		Context("bootstrapping", func() {
+			It("should return error with offlane bootstrap nodes", func() {
```
Comment (jazg) [inline node_test.go]: offline*

**Comment by tok-kkk** [comment]
Thanks @jazg ,typos have been fixed.

**Comment by tok-kkk** [comment]
Add a function for finding node by its ID.

**Pull request merged**